node/demo_contract.go

```diff
@@ -43,41 +43,108 @@ func (node *Node) AddLotteryContract() {
 		types.HomesteadSigner{},
 		priKey)
 	node.DemoContractAddress = crypto.CreateAddress(crypto.PubkeyToAddress(priKey.PublicKey), uint64(0))
+	node.DemoContractPrivateKey = priKey
 	node.addPendingTransactions(types.Transactions{demoContract})
 }
 
 // CreateTransactionForEnterMethod generates transaction for enter method and add it into pending tx list.
-func (node *Node) CreateTransactionForEnterMethod(amount int64, priKey string) {
+func (node *Node) CreateTransactionForEnterMethod(amount int64, priKey string) error {
 	var err error
 	toAddress := node.DemoContractAddress
 
 	abi, err := abi.JSON(strings.NewReader(contracts.LotteryABI))
 	if err != nil {
 		utils.GetLogInstance().Error("Failed to generate staking contract's ABI", "error", err)
+		return err
 	}
 	bytesData, err := abi.Pack(Enter)
 	if err != nil {
 		utils.GetLogInstance().Error("Failed to generate ABI function bytes data", "error", err)
+		return err
 	}
 
 	key, err := crypto.HexToECDSA(priKey)
 	nonce := node.GetNonceOfAddress(crypto.PubkeyToAddress(key.PublicKey))
+	Amount := big.NewInt(amount)
+	Amount = Amount.Mul(Amount, big.NewInt(params.Ether))
 	tx := types.NewTransaction(
 		nonce,
 		toAddress,
 		0,
-		big.NewInt(amount),
+		Amount,
 		params.TxGas*10,
 		nil,
 		bytesData,
 	)
 
 	if err != nil {
 		utils.GetLogInstance().Error("Failed to get private key", "error", err)
+		return err
 	}
 	if signedTx, err := types.SignTx(tx, types.HomesteadSigner{}, key); err == nil {
 		node.addPendingTransactions(types.Transactions{signedTx})
-	} else {
-		utils.GetLogInstance().Error("Unable to call enter method", "error", err)
+		return nil
 	}
+	utils.GetLogInstance().Error("Unable to call enter method", "error", err)
+	return err
+}
+
+// GetResult2 get current players and their balances.
+func (node *Node) GetResult2() (players []string, balances []uint64) {
```

Comment:
GetResult2 actually checks the balances of DemoAccounts which we tried to fund 1000 ether each